src/main/scala/org/allenai/wikitables/WikiTablesSemanticParserCli.scala

```diff
@@ -228,7 +238,17 @@ class WikiTablesSemanticParserCli extends AbstractCli() {
       List()
     }
     
+    // Call .getContext on every example that we'll use during error
+    // evaluation. This preprocesses the corresponding table using
+    // corenlp and (I think) caches the result somewhere in Sempre.
+    // This will happen during the error evaluation of training anyway,
+    // but doing it up-front makes the training timers more useful. 
+    println("Preprocessing context for train/dev evaluation examples.")
+    trainErrorExamples.foreach(x => x.getContext)
```

Comment:
Are you sure this actually caches anything?  You could definitely cache it in `WikiTablesExample`, by changing `def getContext` to `lazy val context`.  Also, another good spot to do `trainErrorExamples.par.foreach(...).seq`, and with the line below.